pkg/simulation/simIteration.go

```diff
@@ -28,7 +29,17 @@ func (sE *SimEnv) simulationLoop(t *infra.Tower) {
 		sE.replaceAgents(t)
 		// t.TowerStateLog(" end of tick")
 		sE.dayInfo.CurrTick++
+
+		//returns if there is a timeout
+		//continously checks every tick since there is no way to kill a goroutine
+		select {
+		default: //nothing
+		case <-ctx.Done():
+			return
+		}
```

Comment:
Is there a reason the default is before the case?

Reply:
Well, I am not sure. It was like this in some template i followed. Will check if it works if i switch them around

Reply:
Yeah, it still works fine with them the usual way around